src/FullScreenDialog/FullScreenDialog.tsx

```diff
@@ -0,0 +1,29 @@
+import { Box, Dialog, DialogContent } from "@mui/material";
+
+import DialogTitle from "../DialogTitle";
+import { FullScreenDialogProps } from "./FullScreenDialog.types";
+import React from "react";
+
+// This component is used to render a dialog that uses the full width (if the condition is true)
+const FullScreenDialog = (props: FullScreenDialogProps) => {
+  const { condition, onClose, children, dialogTitle } = props;
+  // If the condition is true, render a dialog with the specified title and content
+  if (condition) {
+    return (
+      <Dialog maxWidth="xl" fullWidth open>
+        <DialogTitle onClose={onClose}>
+          <Box sx={dialogTitle ? {} : { p: 2 }}>{dialogTitle}</Box>
+        </DialogTitle>
+        <DialogContent dividers>
+          <Box style={{ height: "calc(100vh - 168px)" }}>{children}</Box>
+        </DialogContent>
+      </Dialog>
+    );
+  }
+  // If the condition is false, render the children as-is
+  else {
+    return children;
+  }
+};
```

Comment:
I think it would be better to rename this component to `ConditionalDialog`. Now I've seen the code, it's not just a fullscreen dialog, it conditionally shows a dialog or just renders the children directly. Could be misleading to just call it FullScreenDialog.